### PR TITLE
freeimage/3.18.0: fix build error using msvc 193 with C++20

### PR DESCRIPTION
**Pull request by elvisdukaj**
Trying to build freeimage using MSVC 193 with C++20 results in a buid error. The issue was reported https://github.com/conan-io/conan-center-index/issues/18921.

I am creating a MR for this issue. I submitted also a MR upstream: [issue](https://github.com/WinMerge/freeimage/issues/19) and [MR](https://github.com/WinMerge/freeimage/pull/20).

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/freeimage/all/conandata.yml [hunk 1]
```diff
@@ -18,3 +18,4 @@ patches:
     - patch_file: "patches/012_use-functions-to-override-libtiff-warning-error-handlers.patch"
     - patch_file: "patches/013_use-typedef-as-already-declared.patch"
     - patch_file: "patches/014_no_auto_ptr.patch"
+    - patch_file: "patches/015_fix_msvc_cpp_20.patch"
```
Comment (uilianries) [inline recipes/freeimage/all/conandata.yml:21]:
```suggestion
    - patch_file: "patches/015_fix_msvc_cpp_20.patch"
      patch_type: "official"
      patch_source: "https://github.com/WinMerge/freeimage/pull/20"
      patch_description: "Fix compilation with MSVC and C++20"
```

Please, always add information about the patch, so we track it.

**Review by uilianries — COMMENTED** [review]
Thank you for your quick response to this bug.

**Comment by uilianries** [comment]
@elvisdukaj Could you please ask for access in the issue #4. So the CI can list you as authorized user.

**Review by jcar87 — CHANGES_REQUESTED** [review]
Hi @elvisdukaj - thank you for your contribution and for looking into this.

Please note that your proposed patch https://github.com/WinMerge/freeimage/pull/20 was *not* submitted to the canonical sources of freeimage, but to an unofficial clone.

Please submit or contact the maintainers at https://freeimage.sourceforge.io/. 

Please note that we will not regularly accept patches fixing problems that are in library code - this should be reported and dealt with upstream. If this library is _not_ maintained anymore, we will not be providing an extended life support in Conan Center. Remember that you can always instruct Conan to install a compatible version.

**Comment by elvisdukaj** [comment]
Hi @jcar87  thank you for your feedback!

Well if it’s not appropriate the patch in this context we can close this MR then. I’ll keep it only internally to my company! The project seems not maintained anymore indeed. I wasn’t aware that the GitHub wasn’t the official one 😞

**Comment by jcar87** [comment]
> Well if it’s not appropriate the patch in this context we can close this MR then. I’ll keep it only internally to my company! The project seems not maintained anymore indeed. I wasn’t aware that the GitHub wasn’t the official one 😞

Thanks so much @elvisdukaj  and thanks for your efforts! Keeping this internal is the way to go in these cases. 
I will close the PR. We have no way of validating the proposed changes: they don't run on our CI, and we are not familiar with the codebase of this library to ascertain that this is compatible and working. For libraries that are no longer maintained by their authors, we will consider them unmaintained and won't be maintaining them ourselves either.

Thanks!

**Pull request closed without merging**